sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/models/FileProperties.java

```diff
@@ -4,32 +4,69 @@
 package com.azure.storage.file.models;
 
 import com.azure.core.implementation.util.ImplUtils;
+import com.azure.storage.file.FileSmbProperties;
+
 import java.time.OffsetDateTime;
 import java.util.Map;
 
 /**
  * Contains property information about a File in the storage File service.
  */
 public final class FileProperties {
-    private final String eTag;
-    private final OffsetDateTime lastModified;
-    private final Map<String, String> metadata;
-    private final String fileType;
-    private final Long contentLength;
-    private final String contentType;
-    private final byte[] contentMD5;
-    private final String contentEncoding;
-    private final String cacheControl;
-    private final String contentDisposition;
-    private final OffsetDateTime copyCompletionTime;
-    private final String copyStatusDescription;
-    private final String copyId;
-    private final String copyProgress;
-    private final String copySource;
-    private final CopyStatusType copyStatus;
-    private final Boolean isServerEncrypted;
+    private String eTag;
+    private OffsetDateTime lastModified;
+    private Map<String, String> metadata;
+    private String fileType;
+    private Long contentLength;
+    private String contentType;
+    private byte[] contentMD5;
+    private String contentEncoding;
+    private String cacheControl;
+    private String contentDisposition;
+    private String contentLanguage;
+    private OffsetDateTime copyCompletionTime;
+    private String copyStatusDescription;
+    private String copyId;
+    private String copyProgress;
+    private String copySource;
+    private CopyStatusType copyStatus;
+    private Boolean isServerEncrypted;
+    private FileSmbProperties smbProperties;
+    private String filePermission;
+
+    /**
+     * Creates an empty instance of FileProperties
+     */
+    public FileProperties() {
+    }
+
+    /**
+     * Creates an instance of user settable property information about a specific File.
+     *
+     * @param contentType The Content-Type specified for the file. The default content type is application/octet-stream.
+     * @param contentEncoding The Content-Encoding request header.
+     * @param contentLanguage The Content-Language request header.
+     * @param cacheControl The Cache-Control request header.
+     * @param contentMD5 The MD5 hash of the file to check the message content integrity.
+     * @param contentDisposition The Content-Disposition header that specifies how to process the response.
+     * @param smbProperties The SMB properties of the file.
+     * @param filePermission The file permission of the file. smbProperties.filePermissionKey and filePermission must
+     *                       not both be set simultaneously
+     */
+    public FileProperties(final String contentType, final String contentEncoding, final String contentLanguage,
```

Comment:
I kind of want to add builder style setters for the settable properties to make it easier to build the object

Reply:
Another addition would be to maybe add a setter for the http headers from a filehttpheaders object?

Reply:
There's an open issue on this sdk to update the interfaces for all these helper types to use get and set prefixes. I'd advise that whatever you do is consistent with that so you don't hit merge conflicts/your work isn't undone.

Reply:
Talked to Alan, he said the change will happen after this preview, so it's best to stay consistent with what exists.